src/main/webapp/creatives/maps/google_maps_client.js

```diff
@@ -0,0 +1,252 @@
+import {CreativeMessage} from '../constants.js';
+const DEFAULT_MAP_LAT = 37.422004;
+const DEFAULT_MAP_LNG = -122.081402;
+const DEFAULT_ZOOM = 13;
+const DEFAULT_LOCATION_NUM_DISPLAYED = 4;
+const MARKER_SIZE = 25;
+
+/**
+ * This class handles all map related functionality, including
+ * displaying the map, calculating & displaying directions, tracking all
+ * interactions with the map, logging error messages with SimidProtocol,
+ * and handling any errors that result from the Maps API.
+ */
+export default class GoogleMapsClient {
+    /**
+     * A GoogleMapsClient object handles 
+     * @param {!UserActivityLogger} userSession A User Activity Logger object to track interactions.
+     * @param {!string} query The search query string.
+     * @param {!string} markerUrl The marker image url.
+     * @param {?google.maps.LatLng} coordinates User coordinates.
+     * @param {!SimidProtocol} simidProtocol The SimidProtocol object.
```

Comment:
Make sure you add travelMethod and timeDisplay to the param list

Reply:
Good catch! Added those